File: 2553/separate-the-digits-in-an-array.py
# LeetCode 2553. Separate the Digits in an Array
# 把把數字的每一位數，分開後，再塞進 list 裡
class Solution:
    def separateDigits(self, nums: List[int]) -> List[int]:
        ans = []
        for num in nums:
            for d in str(num):  # 轉換成字串，就能「逐位」取出來
                ans.append(int(d))  # 再轉回數字，塞入答案裡
        return ans

File: 2553/separate-the-digits-in-an-array.cpp
// LeetCode 2553. Separate the Digits in an Array
// 把把數字的每一位數，分開後，再塞進答案裡
class Solution {
public:
    vector<int> separateDigits(vector<int>& nums) {
        vector<int> ans;
        for(int num : nums){
            for(char d : std::to_string(num)){
                ans.push_back(d - '0');
            }
        }
        return ans;
    }
};
